### PR TITLE
Allow permissions change of puppet.conf

### PR DESCRIPTION
**Pull request by teluq-pbrideau**
There is confidential info in my `puppet.conf` (`http_proxy_password` to be precise) that I would like to secure from unprivileged users.

### CODE REVIEW

**Comment by teluq-pbrideau** [comment]
I don’t understand why the check failed…  The two warnings in `manifest/server/config.pp` are not changed I made.  And if I run `bundle exec rake validate lint check` on my laptop, there is about 2000 lines of errors, and none seems to be about my changes.  What do I miss?

**Comment by ekohl** [comment]
I think I need to finish https://github.com/theforeman/puppet-puppet/pull/841 first.

File: manifests/init.pp [hunk 1]
```diff
@@ -752,6 +755,7 @@
   Optional[Stdlib::Absolutepath] $server_versioned_code_id = undef,
   Optional[Stdlib::Absolutepath] $server_versioned_code_content = undef,
   Array[String[1]] $server_jolokia_metrics_whitelist = [],
+  Pattern[/^[0-9]{3,4}$/] $puppetconf_mode = $puppet::params::puppetconf_mode,
```
Comment (ekohl) [inline manifests/init.pp:758]: Would `Stdlib::Filemode` be better?
```suggestion
  Stdlib::Filemode $puppetconf_mode = $puppet::params::puppetconf_mode,
```

Comment (teluq-pbrideau) [reply manifests/init.pp]: Thanks for pointing this out, I was not aware this existed!

**Pull request merged**